src/plugins/telemetryTable/TelemetryTable.js

```diff
@@ -125,25 +135,134 @@ define([
+            const telemetryRemover = this.getTelemetryRemover();
+
+            this.removeTelemetryCollection(keyString);
+
+            this.telemetryCollections[keyString] = this.openmct.telemetry
+                .requestTelemetryCollection(telemetryObject, requestOptions);
+
+            this.telemetryCollections[keyString].on('remove', telemetryRemover);
+            this.telemetryCollections[keyString].on('add', telemetryProcessor);
+            this.telemetryCollections[keyString].load();
+
+            this.decrementOutstandingRequests();
+
+            this.telemetryObjects[keyString] = {
+                telemetryObject,
+                keyString,
+                requestOptions,
+                columnMap,
+                limitEvaluator
+            };
 
             this.emit('object-added', telemetryObject);
         }
 
+        getTelemetryProcessor(keyString, columnMap, limitEvaluator) {
+            return (telemetry) => {
+                //Check that telemetry object has not been removed since telemetry was requested.
+                if (!this.telemetryObjects[keyString]) {
+                    return;
+                }
+
+                let telemetryRows = telemetry.map(datum => new TelemetryTableRow(datum, columnMap, keyString, limitEvaluator));
+
+                if (this.paused) {
+                    this.delayedActions.push(this.tableRows.addRows.bind(this, telemetryRows, 'add'));
+                } else {
+                    this.tableRows.addRows(telemetryRows, 'add');
+                }
+            };
+        }
+
+        getTelemetryRemover() {
+            return (telemetry) => {
+                if (this.paused) {
+                    this.delayedActions.push(this.tableRows.removeRowsByData.bind(this, telemetry));
+                } else {
+                    this.tableRows.removeRowsByData(telemetry);
+                }
+            };
+        }
+
+        /**
+         * @private
+         */
+        incrementOutstandingRequests() {
+            if (this.outstandingRequests === 0) {
+                this.emit('outstanding-requests', true);
+            }
+
+            this.outstandingRequests++;
+        }
+
+        /**
+         * @private
+         */
+        decrementOutstandingRequests() {
+            this.outstandingRequests--;
+
+            if (this.outstandingRequests === 0) {
+                this.emit('outstanding-requests', false);
+            }
+        }
+
+        // will pull all necessary information for all existing bounded telemetry
+        // and pass to table row collection to reset without making any new requests
+        // triggered by filtering
+        resetRowsFromAllData() {
+            let allRows = [];
+
+            Object.keys(this.telemetryCollections).forEach(keyString => {
+                let { columnMap, limitEvaluator } = this.telemetryObjects[keyString];
+
+                this.telemetryCollections[keyString].boundedTelemetry.forEach(datum => {
```

Comment:
I think TelemetryCollections should have a function for accessing the telemetry contained in them. eg. `getAll()`.

Reply:
Agreed, I'll add that!